File: apps/web/lib/documents.ts
export function digitsOnly(value: string): string {
  return (value ?? "").replace(/\D/g, "");
}

export function formatCedula(value: string): string {
  const digits = digitsOnly(value);
  if (!digits) return "";
  return "V-" + digits.replace(/\B(?=(\d{3})+(?!\d))/g, ".");
}

export function formatRif(value: string): string {
  const digits = digitsOnly(value);
  if (!digits) return "";
  if (digits.length <= 1) return "J-" + digits;
  return "J-" + digits.slice(0, -1) + "-" + digits.slice(-1);
}

export function formatPassport(value: string): string {
  const digits = digitsOnly(value);
  if (!digits) return "";
  return "E-" + digits;
}

export function formatDocument(type: string, value: string): string {
  if (!value) return "";
  switch (type) {
    case "ci":
      return formatCedula(value);
    case "rif":
      return formatRif(value);
    case "passport":
      return formatPassport(value);
    default:
      return value;
  }
}

File: apps/api/src/cresko_api/documents.py
import re


def digits_only(value: str | None) -> str:
    if not value:
        return ""
    return re.sub(r"\D", "", value)


def format_ci(value: str | None) -> str:
    digits = digits_only(value)
    if not digits:
        return ""
    grouped = re.sub(r"\B(?=(\d{3})+(?!\d))", ".", digits)
    return f"V-{grouped}"


def format_rif(value: str | None) -> str:
    digits = digits_only(value)
    if not digits:
        return ""
    if len(digits) <= 1:
        return f"J-{digits}"
    return f"J-{digits[:-1]}-{digits[-1]}"


def format_passport(value: str | None) -> str:
    digits = digits_only(value)
    if not digits:
        return ""
    return f"E-{digits}"


def format_document(document_type: str, value: str | None) -> str | None:
    if not value:
        return value
    if document_type == "ci":
        return format_ci(value)
    if document_type == "rif":
        return format_rif(value)
    if document_type == "passport":
        return format_passport(value)
    return value